test/inductor/test_cpu_select_algorithm.py

```diff
@@ -829,6 +834,236 @@ def forward(self, x):
         else:
             self._check_amx_counter(vec_amx)
 
+    @unittest.skipIf(
+        (not issubclass(cpu_vec_isa.pick_vec_isa().__class__, cpu_vec_isa.VecAVX512))
+        and (not issubclass(cpu_vec_isa.pick_vec_isa().__class__, cpu_vec_isa.VecAVX2)),
+        "Only support avx2 or avx512",
+    )
+    @dtypes(torch.bfloat16, torch.half, torch.float)
+    def test_microgemm_fp32_vec(self, dtype):
+        # gemm template of microgemm_fp32_vec
+        TEST_GEMM_TEMPLATE = r"""
+{{GEMM_DEFINE}}
+extern "C" void cpp_micro_gemm(const {{input_type}}* X, const {{input_type}}* W, float* Y) {
+    {{GEMM_CALL}}
+}
+"""
+
+        # gemm template for reference
+        TEST_REF_TEMPLATE = r"""
+#include <torch/csrc/inductor/aoti_torch/c/shim.h>
+template <bool accum>
+inline void kernel_micro_gemm(
+    const {{input_type}}* __restrict__ A,
+    const {{input_type}}* __restrict__ B,
+    float* __restrict__ C,
+    int64_t M,
+    int64_t N,
+    int64_t K,
+    int64_t lda,
+    int64_t ldb,
+    int64_t ldc,
+    bool trans_b) {
+    for (int64_t m = 0; m < M; ++m) {
+        for (int64_t n = 0; n < N; ++n) {
+            float result = accum ? C[m * ldc + n] : 0;
+            for (int64_t k = 0; k < K; ++k) {
+                if (trans_b) {
+                    result += (float)A[m * lda + k] * (float)B[k + n * ldb];
+                } else {
+                    result += (float)A[m * lda + k] * (float)B[k * ldb + n];
+                }
+            }
+            C[m * ldc + n] = result;
+        }
+    }
+}
+extern "C" void cpp_ref_gemm(const {{input_type}}* X, const {{input_type}}* W, float* Y) {
+    {{REF_CALL}}
+}
+"""
+
+        def ref_gemm_codegen_call(
```

Comment:
We do have reference kernel implementation in `cpp_micro_gemm.py`. Can you reuse that instead of redefining things here? Then, we can just use `torch.compile` with different shapes to cover the new transposed B matrix feature you add here.

Reply:
Since micro gemms don't seem to have been exposed and is hard to be called directly by torch.compile, this test case is indeed a bit hacky and exposes a lot of implementation details.
This test case is more to show the correctness.
We plan to remove this test case and add new higher-level test cases by flex attention in the next PR.

Reply:
Since the micro gemm classes are internal implementations, better not directly calling them from the test cases. Can we build pytorch test cases with various shapes and strides to cover your optimized scenarios?

Reply:
Yes, we can build high level test cases with various shapes and strides when testing flex attention and remove the this test case in the next PR.